ext/TenetReactantExt.jl

```diff
@@ -213,6 +214,10 @@ Base.@nospecializeinfer @noinline function Tenet.contract(
     return Tensor(data, ic)
 end
 
+@reactant_overlay function Tenet.contract(@nospecialize(::AbstractBackend), @nospecialize(xs::Tensor...); kwargs...)
+    return reduce((x, y) -> contract(x, y; kwargs...), xs)
+end
```

Comment:
```suggestion
@reactant_overlay function Tenet.contract(@nospecialize(b::AbstractBackend), @nospecialize(xs::Tensor...); kwargs...)
    return reduce((x, y) -> contract(b, x, y; kwargs...), xs)
end
```
I think you might have lost the backend here

Reply:
Yes, that's on purpose because when tracing with Reactant.jl, it uses no backend. On tracing, Reactant.jl just registers that a contraction is performed (i.e. by emitting the `stablehlo.dot_general` MLIR op) but no information of the backend is set. Afterwards, XLA will choose the correct implementation for the backend we want (i.e. CPU, GPU, TPU or distributed).

tldr: Reactant.jl ignores the "backend" and sets its own.

Reply:
Actually, you do a pretty good job and Reactant.jl is slightly slower than TensorOperations.jl, so well done guys :)

but in any case, Reactant.jl is a compiler pipeline and its power is on seeing *all* the contractions collectively, and optimizing all the contractions as a whole.

one cool idea we are working on in Reactant.jl is "kernel injection" (right now only supported for CUDA.jl kernels but will add it support for generic Julia in the future): i.e. you can write a kernel in Julia, inject the LLVM IR in our MLIR and we compile it in such a way that XLA can still compile it.

it would be quite cool to check if we can optionally inject TensorOperations into it, but that will be a future project.